### ISSUE TITLE
Serving keys over HTTP does not guarantee authenticity

### ISSUE THREAD

**Issue by rgrove** [mstoiber/pgp.asc#8]
Love the project! One thing jumped out at me though. Quoting from the current pgpasc.org homepage:

> Simply put your public key in a file called pgp.asc in the root of your server. That way people can find your PGP key by going to http://url.com/pgp.asc.
> 
> This takes less time than to search in a public key server, and absolutely guarantees authenticity.

Unfortunately this is bad advice. Serving keys over HTTP doesn't guarantee authenticity since local malware or an attacker with access to the network between the client and the server can intercept the key in transit and replace it with a modified version.

Ideally, PGP keys should be served over HTTPS to prevent MITM tampering. At a minimum, serving keys over HTTP shouldn't be described as "absolutely guaranteeing authenticity", because there is no such guarantee.

**Comment by mxstbr** [comment]
That is a very valid concern, but nothing this project can have control over. Whenever you connect to a untrusted network (i.e. every network except _maybe_ your own, but even that is a stretch), you are suspect to MITM attacks. HTTPS can sometimes prevent that, but it is absolutely no guarantee. Not only that, but most people with personal websites won't buy SSL certificates for them anyway, as there normally is no need to. 

I will still replace the `http://url.com/pgp.asc` with `https://url.com/pgp.asc` the next time I push changes, as it is good practice. Thanks for the notice!

You are right that "absolutely guaranteeing authenticity" is a bad wording, but I honestly couldn't think of another one. If you have an idea, please feel free to join the discussion in Issue #2 !

**Comment by JamborJan** [comment]
You can refer to https://letsencrypt.org/. The costs aspect will be no longer valid when this is accomplished mid 2015.

**Comment by rgrove** [comment]
CloudFlare also offers free SSL CDN service that's easy to use for personal websites: https://www.cloudflare.com/

> That is a very valid concern, but nothing this project can have control over. 

While pgp.asc certainly can't force people to use HTTPS, it can (and should) provide solid security advice that isn't misleading. The only responsible way to serve a PGP key is over HTTPS, so pgp.asc should make that clear.

**Comment by JamborJan** [comment]
@rgrove absolutely agree. Never start describing a half backed solution. 

Btw. as far as I understood the cloud-flare solution: the communication between user and cloud-flare is encrypted, not between cloudflare and your server. Not quite sure if this is valid but if I'm right thats not a proper solution in my eyes.

**Comment by rgrove** [comment]
> Btw. as far as I understood the cloud-flare solution: the communication between user and cloud-flare is encrypted, not between cloudflare and your server. Not quite sure if this is valid but if I'm right thats not a proper solution in my eyes.

This is correct, and you're right that it's not ideal.

**Comment by iamwebrocker** [comment]
Hi,

thanks for the input. i'm on shared hosting, but my provider allows for ssl with a https://sslsites.de/domainname/..... so in case of the pgp.asc there's no reason not to use ssl.

you are right, we should advocate this more clearly.

another point is; if ppl are using ftp via port 21  - there's no way one can be sure that the key ending up on the webspace is indeed a valid one. i'm very concerned about this false sense of security of "i own my webspace" ... esp on shared hosting where theoretically a old joomla (or wordpress or else) version in someone else's webspace could lead to a write access by unknown evildoers.

so we should make clear that ppl need to really get aware of the things going on.
cheers
tom
[edit: spelling]

**Comment by mxstbr** [comment]
> You can refer to https://letsencrypt.org/. The costs aspect will be no longer valid when this is accomplished mid 2015.

I hadn't heard of that, nice! :+1: 

> it can (and should) provide solid security advice that isn't misleading

True, we really should change the wording. 

> so we should make clear that ppl need to rally get aware of the things going on.

I believe pgp.asc can be more than a website telling you to put your PGP key in the root of your server. (Otherwise I wouldn't have put together this [beginner PGP guide](https://github.com/mstoiber/pgp.asc/issues/5) over the last two days) 

I think proper focus is important here. The main thing should still be to get people to put their key on their server. We can then devote a part of the website to the surrounding elements of uploading your key, and I really want to devote a section to other privacy issues. (E.g. what Curtis Wallen focusses on in [his guide](http://curtiswallen.com/pgp/) I also want to devote a section for total beginners (again, that's why I put together the [guide](https://github.com/mstoiber/pgp.asc/issues/5)). 

Basically, I hope the website looks like this one day:
- [ ] Key in root
  - [ ] Why? 
  - [ ] How?
  - [ ] Problems
    - [ ] MITM w/o SSL
    - [ ] Hosting
      - [ ] External hosting
        - [ ] Shared hosting
        - [ ] Hosting provider
- [ ] Beginners Section
  - [ ] Basic Guide ( #5 )
  - [ ] Other privacy problems
- [ ] Advanced section
  - [ ] Outside Problems (I.e. what Curtis talks about)

We are still far off, but I think with enough contributors this could become reality in the not-so-far future. I think the most important issue right now is getting the copy right and then making a more appealing website. 

EDIT 22:00 (GMT+1): Changed to https on branch gh-pages, see e3e5c5cdc3f67488a021002255aec779a01e71a6
